### ISSUE TITLE
Highlighting issues on Outlook and GMail

### ISSUE THREAD

**Issue by sbrudz** [defmethodinc/just-not-sorry#117]
Tami tested the [v2.0.0-beta.7 release](https://github.com/defmethodinc/just-not-sorry/releases/tag/v2.0.0-beta.7) and found issues with the highlighting on both Outlook and GMail.

From Tami:
> Outlook isn't working right… not everything gets caught but I'm not sure why because some were copy and pasted and fine and others were not and the same goes for typed fresh and new.
![jns-error-1](https://user-images.githubusercontent.com/1520763/114105188-f3080100-9891-11eb-8f6e-0c0425c95704.png)
> For gmail I had a similar experience 
![jsn-error-2](https://user-images.githubusercontent.com/1520763/114105204-fe5b2c80-9891-11eb-925b-d5ece551652b.png)
> I thought the we/i believe/feel worked, but  it seems the We's are missing. Also weirdly when I added a space after the try that doesn't trigger, both it and does that make sense triggered. Here's the copy:
```
sorry i think that just sucks
we believe that trying to actually 
try
does that make sense ?
i believe 
we feel 
i feel 
try 
```

**Comment by sbrudz** [comment]
In testing this fix with Tami, we found that it introduced some new bugs. For example: 
![image](https://user-images.githubusercontent.com/1520763/124990658-45c69580-e006-11eb-9981-aefb44a42f2b.png)
The new, looser rules are picking up "I think" and highlighting it.  Phrases like "Tami feels" and "Our alumni believe" will also get highlights.

**Comment by sbrudz** [comment]
Here are some additional details for future reference.

The situation is that when the email DOM text looks like this:

```
<div>
  just checking
  <div>
    just
  </div>
</div>
```

Both "just" instances should be highlighted, but the bug is that only the first one is being highlighted.

I tracked it down to this line in util.js: https://github.com/defmethodinc/just-not-sorry/blob/beta/src/helpers/util.js#L27
When we call el.textContent, the resulting string is `"just checkingjust"` and the regular expression match for just, `\b(just)\b` doesn't pick up the second one.

I tried using `el.innerText` instead, which returns `"just checking\njust"` The match function now picks up both instances, which is good.  But the highlight is now misaligned because the `\n` line break adds another character, which isn't actually in the DOM.  I added some logic to compensate for this, but found that `textContent` handles text nodes with spaces at the end slightly differently than `innerText` does, causing misalignments that I'm not sure how to compensate for.

I've tried several other options, such as using a TreeWalker, but nothing so far has handled all of the different cases.

**Comment by sbrudz** [comment]
I've created PR #129 to revert the changes in #124 that loosened the logic too much.

The currently released version of JustNotSorry (1.6.3) has this same issue.  Given the difficulty of fixing the bug and the low impact of the problem, I recommend that we stop work on this and focus on other priorities. @tamireiss Are you okay with this course of action?
